airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/partitions/record.py

```diff
@@ -13,9 +13,10 @@ class Record:
     Represents a record read from a stream.
     """
 
-    def __init__(self, data: Mapping[str, Any], partition: "Partition"):
+    def __init__(self, data: Mapping[str, Any], partition: "Partition", is_file_transfer_message: bool = False):
         self.data = data
```

Comment:
Record class acts as an envelop for [the record message](https://github.com/airbytehq/airbyte/blob/9a51bce6f63e099a49f0f24dee9fd9dedf06ef29/airbyte-cdk/python/airbyte_cdk/models/airbyte_protocol.py#L80) so I think is correct to add is_file_transfer_message flag that will match the [stream_data_to_airbyte_message ](https://github.com/airbytehq/airbyte/blob/9a51bce6f63e099a49f0f24dee9fd9dedf06ef29/airbyte-cdk/python/airbyte_cdk/sources/utils/record_helper.py#L15-L21)signature to create the correct message.

That said, I'm not totally opposed to the idea of creating a different class like FileRecord with a file attribute if anybody feels that is not the right approach. However, following the concept that the record and the record message are not the same, just this class with the flag makes sense, IMO.